lib/manager/composer/artifacts.ts

```diff
@@ -40,10 +40,14 @@ export async function updateArtifacts({
     const localPackageFileName = upath.join(config.localDir, packageFileName);
     await fs.outputFile(localPackageFileName, newPackageFileContent);
     const localLockFileName = upath.join(config.localDir, lockFileName);
+    const localAuthFileName = upath.join(cwd, 'auth.json');
     if (config.isLockFileMaintenance) {
       await fs.remove(localLockFileName);
     }
-    const authJson = {};
+    let authJson = {};
+    if (await fs.pathExists(localAuthFileName)) {
```

Comment:
This should also check `COMPOSER_HOME`
> The third way if you want to pre-configure it is via an auth.json file located in your COMPOSER_HOME or besides your composer.json.

Ref;
* https://getcomposer.org/doc/articles/http-basic-authentication.md
* https://getcomposer.org/doc/03-cli.md#composer-home

Reply:
I haven't done this change in this PR, because the main reason I created the PR right now was to not make renovate not overwrite a committed `auth.json` file. The `auth.json` file in `COMPOSER_HOME` would still work as usual for composer because renovate does not do anything to it. This would also be on the host were renovate runs and I wonder if anyone creates an `auth.json` file there. I would rather wait for a specific use case before doing it.

Reply:
I use `COMPOSE_HOME` that's why I mentioned it hah:
> `mkdir -p ~/.composer/ && echo "$COMPOSER_AUTH_CONFIG" > ~/.composer/auth.json`

Your suggestion is to output the file into the build / current working directory?

Reply:
Right, but afaics this is not affected at all by this PR and will continue to work.

Reply:
Ah sorry, I might have completely misunderstood this PR. You're right, global `auth.json` works just fine.